lib/hotel.rb

```diff
@@ -0,0 +1,258 @@
+# hotel.rb
+
+# class Room -- do we need a room class or just a reservation one?
+require 'date'
+require 'awesome_print'
+
+# https://ruby-doc.org/stdlib-2.4.1/libdoc/date/rdoc/Date.html
+
+# a single reservations
+class Reservation
+  # attribute accessor
+  attr_accessor :start_date, :end_date, :room_number, :total_cost, :reservation_id, :discount, :block_id
+
+  # initialize
+  def initialize(start_year, start_month, start_day, end_year, end_month, end_day, room_number, cost, discount, block_id)
+    @start_date = Date.new(start_year, start_month, start_day)
+    @end_date = Date.new(end_year, end_month, end_day)
+    @room_number = room_number
+    @cost = cost
+    @total_cost = @cost*(1-discount)*(@end_date. - @start_date).to_i
+    @reservation_id = @start_date.to_time.to_i / (60 * 60 * 24) + room_number #ID is start date as integer plus room number
+    @discount = discount
+    @block_id = block_id
+  end
+end
+
+# the hotel system
+class HotelSystem
+  # attribute accessor
+  attr_accessor :room_numbers, :blocked_rooms, :date_reservation_hash, :id_reservation_hash, :blocks_hash, :date_block_hash
+
+  # initialize
+  def initialize()
+    @room_numbers = (1..20).to_a #[1, 2, 3, 4, ... 20]
+    @blocked_rooms = []
+    @date_reservation_hash = {} #hash where key is date and value is rooms reserved on that date
+    @id_reservation_hash = {} # has where key is ID and value is Reservation object
+    @date_block_hash = {} #date key, rooms reserved in block
+    @blocks_hash = {} # key = block ID, value = array start, end, room numbers, rooms_booked, discount
+  end
+
+  # make a reservation
+  # updates attributes: id_reservation_hash and date_reservation_hash
+  def make_reservation(start_year, start_month, start_day, end_year, end_month, end_day, room_number, cost, discount, block_id)
+
+    # check what rooms are not reserved
+    available_rooms = self.get_available_rooms(start_year, start_month, start_day, end_year, end_month, end_day)[0]
```

Comment:
Do you need to call `self.get_available_rooms` twice?

Also this `make_reservation` method is getting pretty big, maybe breaking it up into helper methods  or pushing some functionality into the `Reservation` class.